users/serializers.py

```diff
@@ -0,0 +1,77 @@
+import django.contrib.auth.password_validation as password_validation
+from django.contrib.auth.models import User
+from rest_framework import serializers
+from rest_framework.authtoken.models import Token
+
+
+class UserDetailSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username',)
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username', 'password')
+
+    def save(self):
+        user = User.objects.create_user(self.data["username"])
+        user.set_password(self.data["username"])
+        if "email" in self.data:
+            user.email = self.data["email"]
+
+        user.save()
+
+
+class LoginSerializer(serializers.Serializer):
+    username = serializers.CharField(required=True)
+    password = serializers.CharField(required=True)
+
+    def validate_username(self, value):
+        try:
+            User.objects.get(username=value)
+        except Exception as e:
+            raise serializers.ValidationError(detail=e.args[0])
+        return value
+
+    def validate(self, data):
+        user = User.objects.get(username=data["username"])
+        if not user.check_password(data["password"]):
+            raise serializers.ValidationError(detail="Wrong password.")
+
+        data["user"] = user
+        return data
+
+    def to_representation(self, data):
+        token = Token.objects.get(user=data["user"])
+        return {
+            'username': data["username"],
+            'token': token.key
+        }
+
+
+class ChangePasswordSerializer(serializers.Serializer):
+    old_password = serializers.CharField(required=True)
+    new_password = serializers.CharField(required=True)
+
+    def save(self, user):
+        user.set_password(self.data["new_password"])
+        user.save()
+
+    def validate_new_password(self, value):
+        try:
+            password_validation.validate_password(value)
```

Comment:
You can set `validators` to serializer field 
For example: serializers.CharField(required=True, validators=[validator])